core/src/main/java/com/scalar/db/api/StorageInfo.java

```diff
@@ -0,0 +1,62 @@
+package com.scalar.db.api;
+
+public interface StorageInfo {
+  /**
+   * Returns the storage name.
+   *
+   * @return the storage name
+   */
+  String getStorageName();
+
+  /**
+   * Returns the mutation atomicity unit of the storage.
+   *
+   * @return the mutation atomicity unit of the storage
+   */
+  MutationAtomicityUnit getMutationAtomicityUnit();
```

Comment:
```suggestion
  MutationAtomicityGranularity getMutationAtomicityGranularity();
```
After some thought, I feel it's probably more proper to call it `granularity`, although it's a bit long.
What do you think?

Reply:
Gemini says `MutationAtomicityUnit` is more appropriate, but I'm not sure.

```
In the context of transactions, when referring to the scope within which updates can be performed atomically, **"MutationAtomicityUnit"** is the better choice.

Here's why:

* **Unit**: "Unit" directly implies a distinct, defined **scope** or **entity**. The example you provided with `RECORD`, `PARTITION`, `TABLE`, etc., clearly defines the atomic "unit" of operation. It naturally conveys the idea that mutations are performed atomically for *each* of these specified units.
* **Granularity**: "Granularity" refers to the **fineness or coarseness** of something. While related, it describes the *degree* of detail or division rather than the specific scope itself. For instance, you might discuss whether the atomicity has "fine-grained" or "coarse-grained" granularity. When you want to name the actual *thing* that is atomic, "Unit" is more precise and less ambiguous.

Therefore, "MutationAtomicityUnit" more accurately and clearly communicates the intended meaning of the atomic **scope** for mutations.
```

What do you think? 🙇

Reply:
I also asked Gemini. It suggests Granularity.... I think it depends on how we ask.

**Question:**
I'm considering how to name a method, and I would appreciate your advice.

We provide a transaction manager for various kinds of databases. The transaction manager pushes down write operations into the underlying databases based on the size that the databases can atomically write.

To make each database provide the size information, I created a `StorageInfo` interface, created a class that implements the interface for each database, and created a method that returns the size information.
The size information is an Enum class that has the following values: STORAGE, NAMESPACE, TABLE, PARTITION, and RECORD.

I have two candidate names for the method.
One is `getMutationAtomicityUnit()` and the other is `getMutationAtomicityGranularity()`.
Which do you think is better?
Do you have any other better ideas?

**Answer:**
I think getMutationAtomicityGranularity() is the better choice.

Here’s a breakdown of why, along with some alternative ideas.

Comparison of Your Candidates
Both of your proposed names are very good because they are descriptive and unambiguous. However, getMutationAtomicityGranularity() has a slight edge.

getMutationAtomicityGranularity(): Slightly better. 👍
In database and systems design, granularity is the standard term for referring to the scale or level at which an operation occurs (e.g., fine-grained vs. coarse-grained). Since your Enum defines a scale from RECORD (fine-grained) to STORAGE (coarse-grained), this name is a perfect technical fit.

getMutationAtomicityUnit(): Also very good.
The word unit is also accurate. It correctly describes that the method returns the single, indivisible "chunk" for atomic writes. It's simple and clear. The only minor drawback is that "granularity" is more idiomatic in this specific technical domain.

Other Ideas
If you're open to other suggestions, here are a couple of alternatives that aim for simplicity while remaining clear.

getAtomicityLevel()
This is a strong contender. It's shorter and replaces the more technical "Granularity" with the simpler "Level," which is easily understood by any developer. The context of your transaction manager implies that the atomicity relates to mutations.

getAtomicWriteScope()
Using Scope is another excellent option. It clearly communicates that the Enum return value defines the boundary or scope (e.g., the scope of a table, the scope of a partition) within which writes are atomic.

Final Recommendation
Go with getMutationAtomicityGranularity(). It's the most precise and idiomatic term for your use case.

If you feel it's too much jargon, getAtomicityLevel() is a fantastic, simpler alternative that conveys the same meaning effectively.

Reply:
My question was as follows:

In the context of transactions, which term is more appropriate to express the range in which updates can be performed atomically: MutationAtomicityUnit or MutationAtomicityGranularity?



Below is an example using MutationAtomicityUnit:

```java
  /**
   * The mutation atomicity unit of the storage.
   *
   * <p>This enum defines the atomicity unit for mutations in the storage. It determines the scope
   * of atomicity for mutations such as put and delete.
   */
  enum MutationAtomicityUnit {
    /**
     * The atomicity unit is at the record level, meaning that mutations are performed atomically
     * for each record.
     */
    RECORD,

    /**
     * The atomicity unit is at the partition level, meaning that mutations are performed atomically
     * for each partition.
     */
    PARTITION,

    /**
     * The atomicity unit is at the table level, meaning that mutations are performed atomically for
     * each table.
     */
    TABLE,

    /**
     * The atomicity unit is at the namespace level, meaning that mutations are performed atomically
     * for each namespace.
     */
    NAMESPACE,

    /**
     * The atomicity unit is at the storage level, meaning that mutations are performed atomically
     * for the entire storage.
     */
    STORAGE
  }
```

Reply:
Based on the discussion, since `Unit` is also correct and simpler, let's go with as is.